lewis/examples/modbus_device/__init__.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# *********************************************************************
+# lewis - a library for creating hardware device simulators
+# Copyright (C) 2016 European Spallation Source ERIC
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# *********************************************************************
+
+from lewis.devices import Device
+from lewis.adapters.modbus import ModbusAdapter, ModbusDataBank
+
+
+class ModbusDevice(Device):
+    pass
+
+
+class ModbusInterface(ModbusAdapter):
+    """
+    The class attributes di, co, ir and hr represent Discrete Inputs, Coils, Input Registers and
+    Holding Registers, respectively. Each attribute should be assigned a ModbusDataBank instance
+    by the Interface implementation.
+
+    Here, two basic ModbusDataBanks are created and initialized to a default value across the full
+    range of valid addresses. One DataBank is shared by di and co, and the other by ir and hr to
+    demonstrate overlaid memory segments. If you want each segment to have its own memory, just
+    create separate instances for all four.
+    """
+    di = ModbusDataBank.create_basic(False)
+    co = di
+    ir = ModbusDataBank.create_basic(0)
+    hr = ir
```

Comment:
Well, without an example device and someone to ask questions about it, any sort of bindings I could implement now would be based entirely on speculation about how I _think_ a modbus device _might_ work (based mostly on how EPICS and TCP Stream devices work). The `nanodac` (so far) only requires this sort of read/write memory functionality. So I'm not sure "urgently" is quite right... Even if we had bindings, we would have no device to use them, which is the reason we don't have bindings to begin with. It's a chicken or the egg type of problem.

Basically, it boils down to: We had no modbus adapter, so Dom complained and we implemented what he needed. Now we have a simple modbus adapter without bindings. We can either A) Wait for someone to complain they're missing and implement them then, or B) Implement speculative bindings and wait for someone to complain they're bad and implement them properly then. I'm not sure which would be better.

Hypothetically, if you were desperate enough, you could implement device behaviour by overriding `handle` in the interface and updating the DataBanks against the Device, and vice versa, every cycle. But that isn't really the "proper" way of doing things, and so I didn't want to encourage it just to make the example a bit more interesting. Seemed better to just show existing functionality and leave future functionality to the future.

Good point about the docs and release notes, thanks for the reminder!